compiler/util/misc.cpp

```diff
@@ -609,9 +609,10 @@ static bool printErrorHeader(BaseAST* ast, astlocT astloc) {
     }
   }
 
-  bool guess = filename && !have_ast_line;
+  int guess = -1;  // -1=no filename:line# printed; 0=not guessed; 1=guessed
 
   if (filename) {
+    guess = !have_ast_line;
```

Comment:
I don't think it's guaranteed that `!false == 1`. So this line needs to be something like `guess = !have_ast_line ? 1 : 0; // The explicit value 1 is actually important here`

Reply:
I don't think that's right...  This SO post seems to back that sense up:  https://stackoverflow.com/questions/5369770/bool-to-int-conversion#:~:text=bool%20values%20are%20convertible%20to,is%20guaranteed%20by%20the%20language.  Could you be thinking of the fact that non-1 `int` values are also considered `true`?

Reply:
I verified that you're correct. I think it was a memory from old times with C, but even that may be mis-remembering.